### 필수과제 2
- 이진 분류할 데이터를 공유할 예정
- 해당 데이터를 가지고 필수과제 1에서 서칭한 파라미터들을 튜닝하면서 모델링을 진행해주세요!
- 데이터 전처리를 진행해야합니다.
- 데이터 전처리 방식도 정리해서 올려주셔야 합니다.
- 실제 데이터 전처리를 최소 2~3개 정도 다르게 진행해 보시면서 (변수를 빼거나 할 수 있음)
- 실제 값이 얼마나 차이가 나는지 확인해주세요!
- 다음 시간에 랜덤으로 또 발표를 진행할 예정 (랜덤 발표가 어려운 경우는 우수학회원이 어렵다.)

In [1]:
import pandas as pd
import numpy as np

In [2]:
movie = pd.read_csv("movie_rv.csv")

In [3]:
movie

Unnamed: 0        id  \
0                0   9976970   
1                1   3819312   
2                2  10265843   
3                3   9045019   
4                4   6483659   
...            ...       ...   
149995      149995   6222902   
149996      149996   8549745   
149997      149997   9311800   
149998      149998   2376369   
149999      149999   9619869   

                                                 document  label  
0                                     아 더빙.. 진짜 짜증나네요 목소리      0  
1                       흠...포스터보고 초딩영화줄....오버연기조차 가볍지 않구나      1  
2                                       너무재밓었다그래서보는것을추천한다      0  
3                           교도소 이야기구먼 ..솔직히 재미는 없다..평점 조정      0  
4       사이몬페그의 익살스런 연기가 돋보였던 영화!스파이더맨에서 늙어보이기만 했던 커스틴 ...      1  
...                                                   ...    ...  
149995                                인간이 문제지.. 소는 뭔죄인가..      0  
149996                                      평점이 너무 낮아서...      1  
149997                    이게 뭐요? 한국인은 거들먹거리고 필리핀 혼혈은 착하다?      0  
149998                        청춘 영화의 최고봉.방황과 우울했던 날들의 자화상      1  
149999                           한국 영화 최초로 수간하는 내용이 담긴 영화      0  

[150000 rows x 4 columns]

### 데이터 전처리

In [4]:
movie.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150000 entries, 0 to 149999
Data columns (total 4 columns):
 #   Column      Non-Null Count   Dtype 
---  ------      --------------   ----- 
 0   Unnamed: 0  150000 non-null  int64 
 1   id          150000 non-null  int64 
 2   document    149995 non-null  object
 3   label       150000 non-null  int64 
dtypes: int64(3), object(1)
memory usage: 4.6+ MB


In [5]:
# Unnamed: 0 (index), id 컬럼 제거
movie = movie.drop(["Unnamed: 0", "id"], axis=1, inplace=False)

In [6]:
# 결측치 확인
movie.isnull().sum()

document    5
label       0
dtype: int64

In [7]:
# 결측치 처리
movie = movie.fillna(" ")

In [8]:
# 특수 기호 제거
movie["document"] = movie["document"].str.replace("[\(.+?\)]*", "")

C:\Users\doram\AppData\Local\Temp\ipykernel_18492\4211706900.py:2: FutureWarning: The default value of regex will change from True to False in a future version.
  movie["document"] = movie["document"].str.replace("[\(.+?\)]*", "")


In [9]:
# 문자열 길이 (length) 추가
movie["length"] = movie["document"].str.len()

In [10]:
movie

document  label  length
0                                       아 더빙 진짜 짜증나네요 목소리      0      17
1                              흠포스터보고 초딩영화줄오버연기조차 가볍지 않구나      1      26
2                                       너무재밓었다그래서보는것을추천한다      0      17
3                               교도소 이야기구먼 솔직히 재미는 없다평점 조정      0      25
4       사이몬페그의 익살스런 연기가 돋보였던 영화!스파이더맨에서 늙어보이기만 했던 커스틴 ...      1      61
...                                                   ...    ...     ...
149995                                    인간이 문제지 소는 뭔죄인가      0      15
149996                                         평점이 너무 낮아서      1      10
149997                      이게 뭐요 한국인은 거들먹거리고 필리핀 혼혈은 착하다      0      29
149998                         청춘 영화의 최고봉방황과 우울했던 날들의 자화상      1      26
149999                           한국 영화 최초로 수간하는 내용이 담긴 영화      0      24

[150000 rows x 3 columns]

In [11]:
# 주요 단어 찾기
import _jpype
from konlpy.tag import Okt

okt = Okt()

In [12]:
def get_all_words(tokenized_text):
    tokenized_filtered = []
    
    for i in tokenized_text:
        for word in i:
            tokenized_filtered.append(word)
    
    return tokenized_filtered

In [13]:
import operator

def word_count(tokenized_data):
    word_counter = dict()
    
    for i in tokenized_data:
        if i in list(word_counter.keys()):
            word_counter[i] += 1
        else:
            word_counter[i] = 1
    
    sorted_dict = dict(sorted(word_counter.items(), key=operator.itemgetter(1), reverse=True))
    
    return sorted_dict

In [14]:
comment_words = ''
tokenized_text = []
for val in movie["document"]:

    val = str(val)
    tokens = val.split()
    tokenized_text.append(tokens)
    comment_words += " ".join(tokens) + " "

In [15]:
all_words = get_all_words(tokenized_text)

In [16]:
tokenized_dict = word_count(all_words)
# 인터넷에 있는 분석으로 대체

KeyboardInterrupt: 

[('최고의', 2219),
 ('ㅋㅋ', 2019),
 ('좋은', 1726),
 ('재밌게', 1343),
 ('쓰레기', 1295),
 ('아', 1204),
 ('드라마', 1186),
 ('꼭', 1181),
 ('가장', 1178),
 ('보면', 1176),
 ('ㅋㅋㅋ', 1108),
 ('마지막', 1070),
 ('스토리', 1020),
 ('무슨', 999),
 ('ㅋ', 989),
 ('없고', 981),
 ('.', 981),
 ('볼', 959),
 ('ㅠㅠ', 958),
 ('같은', 956)]

In [18]:
from sklearn.preprocessing import LabelEncoder
import re

label_encoder = LabelEncoder()

In [62]:
# 주요 단어에 대한 컬럼 추가
movie["temp"] = movie["document"].str.extract("(최고)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["최고"] = temp["temp_name"]

In [67]:
movie[145:150]

document  label  length temp  \
145                                       용가리 진짜짱짱맨이다ㅋ      0      12  NaN   
146  이 영화를 이제서야 보다니 감히 내 인생 최고의 영화중 하나로 꼽을 수 있을만한 작...      1     110   최고   
147      작가가별로다 내용이진짜별로임 맨날그냥기대하고 재방송하면 혹시나하고봐줘도 답없다진짜      0      45  NaN   
148                명작도 이런 명작이 있을까 싶다 보고 또 봐도 여운이 남는 영화      1      35  NaN   
149  아~ 진짜 조금만 더 손 좀 보면 왠만한 상업 영화 못지 않게 퀄리티 쩔게 만들어 ...      1     140  NaN   

     KK  최고  
145   0   0  
146   0   1  
147   0   0  
148   0   0  
149   0   0

In [101]:
movie["temp"] = movie["document"].str.extract("(?<=[^ㅋ])(ㅋㅋ)(?=[^ㅋ])")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["KK"] = temp["temp_name"]

In [70]:
movie["temp"] = movie["document"].str.extract("(좋은)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["좋은"] = temp["temp_name"]

In [73]:
movie["temp"] = movie["document"].str.extract("(좋은데)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["좋은데"] = temp["temp_name"]

In [80]:
movie["temp"] = movie["document"].str.extract("(재밌게)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["재밌게"] = temp["temp_name"]

In [83]:
movie["temp"] = movie["document"].str.extract("(쓰레기)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["쓰레기"] = temp["temp_name"]

In [87]:
movie["temp"] = movie["document"].str.extract("(드라마)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["드라마"] = temp["temp_name"]

In [90]:
movie["temp"] = movie["document"].str.extract("(꼭)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["꼭"] = temp["temp_name"]

In [94]:
movie["temp"] = movie["document"].str.extract("(가장)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["가장"] = temp["temp_name"]

In [97]:
movie["temp"] = movie["document"].str.extract("(보면)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["보면"] = temp["temp_name"]

In [128]:
movie["temp"] = movie["document"].str.extract("(?<=[^ㅋ])(ㅋㅋㅋ)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["KKK_more"] = temp["temp_name"]

In [131]:
movie["temp"] = movie["document"].str.extract("(마지막)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["마지막"] = temp["temp_name"]

In [134]:
movie["temp"] = movie["document"].str.extract("(스토리)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["스토리"] = temp["temp_name"]

In [137]:
movie["temp"] = movie["document"].str.extract("(무슨)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["무슨"] = temp["temp_name"]

In [140]:
movie["temp"] = movie["document"].str.extract("(?<=[^ㅋ])(ㅋ)(?=[^ㅋ])")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["K"] = temp["temp_name"]

In [143]:
movie["temp"] = movie["document"].str.extract("(없고)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["없고"] = temp["temp_name"]

In [146]:
movie["temp"] = movie["document"].str.extract("(ㅠ)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["so_sad"] = temp["temp_name"]

In [166]:
movie["temp"] = movie["document"].str.extract("(짜증)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["짜증"] = temp["temp_name"]

In [170]:
movie["temp"] = movie["document"].str.extract("(추천)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["추천"] = temp["temp_name"]

In [174]:
movie["temp"] = movie["document"].str.extract("(연기)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["연기"] = temp["temp_name"]

In [179]:
movie["temp"] = movie["document"].str.extract("(배우)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["배우"] = temp["temp_name"]

In [183]:
movie["temp"] = movie["document"].str.extract("(감독)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["감독"] = temp["temp_name"]

In [210]:
movie["temp"] = movie["document"].str.extract("(OO)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["bad_words"] = temp["temp_name"]

In [215]:
movie["temp"] = movie["document"].str.extract("(♥)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["heart"] = temp["temp_name"]

In [219]:
movie["temp"] = movie["document"].str.extract("(원작)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["원작"] = temp["temp_name"]

In [223]:
movie["temp"] = movie["document"].str.extract("(망)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["망"] = temp["temp_name"]

In [230]:
movie["temp"] = movie["document"].str.extract("(?<=[ ])(명)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["명"] = temp["temp_name"]

In [234]:
movie["temp"] = movie["document"].str.extract("(평점)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["평점"] = temp["temp_name"]

In [247]:
movie["temp"] = movie["document"].str.extract("(재미)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["재미"] = temp["temp_name"]

In [249]:
movie["temp"] = movie["document"].str.extract("(재미없)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["재미없"] = temp["temp_name"]

In [261]:
movie["temp"] = movie["document"].str.extract("(절대)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["절대"] = temp["temp_name"]

In [265]:
movie["temp"] = movie["document"].str.extract("(문제)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["문제"] = temp["temp_name"]

In [268]:
movie["temp"] = movie["document"].str.extract("(좀)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["좀"] = temp["temp_name"]

In [271]:
movie["temp"] = movie["document"].str.extract("(조금)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["조금"] = temp["temp_name"]

In [274]:
movie["temp"] = movie["document"].str.extract("(별로)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["별로"] = temp["temp_name"]

In [277]:
movie["temp"] = movie["document"].str.extract("(감동)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["감동"] = temp["temp_name"]

In [285]:
movie["temp"] = movie["document"].str.extract("(있는)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["있는"] = temp["temp_name"]

In [286]:
movie["temp"] = movie["document"].str.extract("(없는)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["없는"] = temp["temp_name"]

In [293]:
movie["temp"] = movie["document"].str.extract("(다시)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["다시"] = temp["temp_name"]

In [296]:
movie["temp"] = movie["document"].str.extract("(참)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["참"] = temp["temp_name"]

In [299]:
movie["temp"] = movie["document"].str.extract("(또)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["또"] = temp["temp_name"]

In [308]:
movie["temp"] = movie["document"].str.extract("(높)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["높"] = temp["temp_name"]

In [311]:
movie["temp"] = movie["document"].str.extract("(낮)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["낮"] = temp["temp_name"]

In [319]:
movie["temp"] = movie["document"].str.extract("(보다)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["보다"] = temp["temp_name"]

In [322]:
movie["temp"] = movie["document"].str.extract("(지루)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["지루"] = temp["temp_name"]

In [328]:
movie["temp"] = movie["document"].str.extract("(돈)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["돈"] = temp["temp_name"]

In [338]:
movie["temp"] = movie["document"].str.extract("(존나)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["존나"] = temp["temp_name"]

In [344]:
movie["temp"] = movie["document"].str.extract("(!)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["exclaim"] = temp["temp_name"]

In [350]:
movie["temp"] = movie["document"].str.extract("(~)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["wave"] = temp["temp_name"]

In [353]:
movie["temp"] = movie["document"].str.extract("(시간)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["시간"] = temp["temp_name"]

In [356]:
movie["temp"] = movie["document"].str.extract("(전기)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["전기"] = temp["temp_name"]

In [359]:
movie["temp"] = movie["document"].str.extract("(아깝)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["아깝"] = temp["temp_name"]

In [365]:
movie["temp"] = movie["document"].str.extract("(모르)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["모르"] = temp["temp_name"]

In [368]:
movie["temp"] = movie["document"].str.extract("(행복)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["행복"] = temp["temp_name"]

In [371]:
movie["temp"] = movie["document"].str.extract("(괜찮)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["괜찮"] = temp["temp_name"]

In [374]:
movie["temp"] = movie["document"].str.extract("(정말)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["정말"] = temp["temp_name"]

In [377]:
movie["temp"] = movie["document"].str.extract("(매우)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["매우"] = temp["temp_name"]

In [380]:
movie["temp"] = movie["document"].str.extract("(완전)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["완전"] = temp["temp_name"]

In [383]:
movie["temp"] = movie["document"].str.extract("(짱)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["짱"] = temp["temp_name"]

In [394]:
movie["temp"] = movie["document"].str.extract("(오랜)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["오랜"] = temp["temp_name"]

In [406]:
movie["temp"] = movie["document"].str.extract("(관객)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["관객"] = temp["temp_name"]

In [413]:
movie["temp"] = movie["document"].str.extract("(억지)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["억지"] = temp["temp_name"]

In [417]:
movie["temp"] = movie["document"].str.extract("(눈물)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["눈물"] = temp["temp_name"]

In [426]:
movie["temp"] = movie["document"].str.extract("(이런)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["이런"] = temp["temp_name"]

In [429]:
movie["temp"] = movie["document"].str.extract("(이렇게)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["이렇게"] = temp["temp_name"]

In [444]:
movie["temp"] = movie["document"].str.extract("(사랑)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["사랑"] = temp["temp_name"]

In [450]:
movie["temp"] = movie["document"].str.extract("(우리)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["우리"] = temp["temp_name"]

In [453]:
movie["temp"] = movie["document"].str.extract("(이게)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["이게"] = temp["temp_name"]

In [456]:
movie["temp"] = movie["document"].str.extract("(뭐)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["뭐"] = temp["temp_name"]

In [459]:
movie["temp"] = movie["document"].str.extract("(뭔)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["뭔"] = temp["temp_name"]

In [468]:
movie["temp"] = movie["document"].str.extract("(좋)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["좋"] = temp["temp_name"]

In [471]:
movie["temp"] = movie["document"].str.extract("(싫)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["싫"] = temp["temp_name"]

In [474]:
movie["temp"] = movie["document"].str.extract("(매력)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["매력"] = temp["temp_name"]

In [480]:
movie["temp"] = movie["document"].str.extract("(기대)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["기대"] = temp["temp_name"]

In [504]:
movie["temp"] = movie["document"].str.extract("(너무)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["너무"] = temp["temp_name"]

In [514]:
movie["temp"] = movie["document"].str.extract("(내용)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["내용"] = temp["temp_name"]

In [517]:
movie["temp"] = movie["document"].str.extract("(이해)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["이해"] = temp["temp_name"]

In [524]:
movie["temp"] = movie["document"].str.extract("(실망)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["실망"] = temp["temp_name"]

In [530]:
movie["temp"] = movie["document"].str.extract("(여운)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["여운"] = temp["temp_name"]

In [537]:
movie["temp"] = movie["document"].str.extract("(장난)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["장난"] = temp["temp_name"]

In [544]:
movie["temp"] = movie["document"].str.extract("(그냥)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["그냥"] = temp["temp_name"]

In [550]:
movie["temp"] = movie["document"].str.extract("(후속)")
label_encoder.fit(movie["temp"])
labels = label_encoder.transform(movie["temp"])
temp = pd.DataFrame(data=labels, columns=["temp_name"])
temp["temp_name"].replace({0:1, 1:0}, inplace=True)
movie["후속"] = temp["temp_name"]

In [553]:
#movie.후속.value_counts()
#movie.query("그냥 == 1").label.value_counts()
#movie.drop(["후속"], axis=1, inplace=True)

In [1]:
#no_info = movie.query("KK == 0 and 좋은 == 0 and 재밌게 == 0 and 쓰레기 == 0 and 드라마 == 0 and 꼭 == 0 and 가장 == 0 and 보면 == 0 and KKK_more == 0 and 마지막 == 0 and 스토리 == 0 and 무슨 == 0 and K == 0 and 없고 == 0 and so_sad == 0 and 짜증 == 0 and 추천 == 0 and 배우 == 0 and 연기 == 0 and 감독 == 0 and bad_words == 0 and heart == 0 and 원작 == 0 and 망 == 0 and 명 == 0 and 평점 == 0 and 재미 == 0")
#no_info2 = no_info.query("절대 == 0 and 문제 == 0 and 좀 == 0 and 조금 == 0 and 별로 == 0 and 감동 == 0 and 있는 == 0 and 없는 == 0 and 다시 == 0 and 참 == 0 and 또 == 0 and 높 == 0 and 낮 == 0 and 보다 == 0 and 지루 == 0 and 돈 == 0 and 존나 == 0 and exclaim == 0 and wave == 0 and 시간 == 0 and 전기 == 0 and 아깝 == 0 and 모르 == 0 and 행복 == 0 and 괜찮 == 0 and 정말 == 0 and 매우 == 0 and 완전 == 0 and 짱 == 0 and 오랜 == 0 and 관객 == 0")
#no_info2.query("최고 == 0 and 억지 == 0 and 눈물 == 0 and 이런 == 0 and 이렇게 == 0 and 사랑 == 0 and 우리 == 0 and 이게 == 0 and 뭐 == 0 and 뭔 == 0 and 좋 == 0 and 싫 == 0 and 매력 == 0 and 기대 == 0 and 너무 == 0 and 내용 == 0 and 이해 == 0 and 실망 == 0 and 여운 == 0 and 장난 == 0 and 그냥 == 0").document

In [555]:
movie

document  label  length temp  \
0                                       아 더빙 진짜 짜증나네요 목소리      0      17  NaN   
1                              흠포스터보고 초딩영화줄오버연기조차 가볍지 않구나      1      26  NaN   
2                                       너무재밓었다그래서보는것을추천한다      0      17  NaN   
3                               교도소 이야기구먼 솔직히 재미는 없다평점 조정      0      25  NaN   
4       사이몬페그의 익살스런 연기가 돋보였던 영화!스파이더맨에서 늙어보이기만 했던 커스틴 ...      1      61  NaN   
...                                                   ...    ...     ...  ...   
149995                                    인간이 문제지 소는 뭔죄인가      0      15  NaN   
149996                                         평점이 너무 낮아서      1      10  NaN   
149997                      이게 뭐요 한국인은 거들먹거리고 필리핀 혼혈은 착하다      0      29  NaN   
149998                         청춘 영화의 최고봉방황과 우울했던 날들의 자화상      1      26  NaN   
149999                           한국 영화 최초로 수간하는 내용이 담긴 영화      0      24  NaN   

        KK  최고  좋은  좋은데  재밌게  쓰레기  ...  싫  매력  기대  너무  내용  이해  실망  여운  장난  그냥  
0        0   0   0    0    0    0  ...  0   0   0   0   0   0   0   0   0   0  
1        0   0   0    0    0    0  ...  0   0   0   0   0   0   0   0   0   0  
2        0   0   0    0    0    0  ...  0   0   0   1   0   0   0   0   0   0  
3        0   0   0    0    0    0  ...  0   0   0   0   0   0   0   0   0   0  
4        0   0   0    0    0    0  ...  0   0   0   1   0   0   0   0   0   0  
...     ..  ..  ..  ...  ...  ...  ... ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  
149995   0   0   0    0    0    0  ...  0   0   0   0   0   0   0   0   0   0  
149996   0   0   0    0    0    0  ...  0   0   0   1   0   0   0   0   0   0  
149997   0   0   0    0    0    0  ...  0   0   0   0   0   0   0   0   0   0  
149998   0   1   0    0    0    0  ...  0   0   0   0   0   0   0   0   0   0  
149999   0   0   0    0    0    0  ...  0   0   0   0   1   0   0   0   0   0  

[150000 rows x 86 columns]

In [562]:
# temp, document 제거
new_movie = movie.drop(["temp", "document"], axis=1, inplace=False)

### 데이터 분할

In [563]:
from sklearn.model_selection import train_test_split

In [564]:
y = np.array(new_movie["label"])
X = new_movie.drop(["label"], axis=1, inplace=False)

In [574]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=111)

### 로지스틱 회귀

In [575]:
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, roc_auc_score

In [576]:
lr_clf = LogisticRegression()
lr_clf.fit(X_train, y_train)
lr_preds = lr_clf.predict(X_test)

print(accuracy_score(y_test, lr_preds), "정확도")
print(roc_auc_score(y_test, lr_preds), "roc")

0.7094222222222222 정확도
0.7089463495575222 roc


C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


### 하이퍼파라미터 찾기

In [578]:
solvers = ["lbfgs", "liblinear", "newton-cg", "sag", "saga"]

for solver in solvers:
    lr_clf = LogisticRegression(solver=solver, max_iter=500)
    lr_clf.fit(X_train, y_train)
    lr_preds = lr_clf.predict(X_test)
    
    print("solver : {0}, accuracy : {1:.3f}, roc_auc : {2:.3f}".format(solver, accuracy_score(y_test, lr_preds), 
                                                                       roc_auc_score(y_test, lr_preds)))

solver : lbfgs, accuracy : 0.710, roc_auc : 0.710
solver : liblinear, accuracy : 0.710, roc_auc : 0.710
solver : newton-cg, accuracy : 0.710, roc_auc : 0.710


C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_sag.py:352: ConvergenceWarning: The max_iter was reached which means the coef_ did not converge
  warnings.warn(


solver : sag, accuracy : 0.710, roc_auc : 0.710
solver : saga, accuracy : 0.710, roc_auc : 0.710


C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_sag.py:352: ConvergenceWarning: The max_iter was reached which means the coef_ did not converge
  warnings.warn(


- solver의 차이는 유의미하지 않음

In [579]:
from sklearn.model_selection import GridSearchCV

def log_tunning(train, y, params):
    lr_clf = LogisticRegression(random_state=111)
    grid = GridSearchCV(lr_clf, params, scoring="roc_auc", cv=7)
    grid.fit(train, y)
    
    print(grid.best_params_)
    print(grid.best_score_)
    
    return grid.best_estimator_

In [580]:
# 파라미터 하나씩 정의해보기 (1)
param1 = {"penalty":["l2","l1"], "C":[0.01, 0.1, 1, 5, 10, 100], "max_iter":[100, 500]}
log_tunning(X_train, y_train, params=param1)

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_

{'C': 100, 'max_iter': 500, 'penalty': 'l2'}
0.7925916245969248


LogisticRegression(C=100, max_iter=500, random_state=111)

In [581]:
# 파라미터 하나씩 정의해보기 (2)
param2 = {"penalty":["l2"], "C":[0.1, 0.2, 0.3, 0.4, 0.5, 1, 5, 10], "max_iter":[100, 200, 300, 400, 500, 700]}
log_tunning(X_train, y_train, params=param2)

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_

{'C': 10, 'max_iter': 300, 'penalty': 'l2'}
0.7926064741396458


C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


LogisticRegression(C=10, max_iter=300, random_state=111)

In [583]:
final_param = {'C':[10], 'max_iter':[300], 'penalty':['l2']}
model = log_tunning(X_train, y_train, params=final_param)
y_pred = model.predict_proba(X_test)

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_

{'C': 10, 'max_iter': 300, 'penalty': 'l2'}
0.7926064741396458


C:\ProgramData\Anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


In [584]:
y_pred

array([[0.84966103, 0.15033897],
       [0.26423978, 0.73576022],
       [0.53778223, 0.46221777],
       ...,
       [0.76025727, 0.23974273],
       [0.29997494, 0.70002506],
       [0.40022295, 0.59977705]])

- 파라미터 값의 조정을 통해 정확도를 0.71에서 0.79까지 높일 수 있었다.